src/data.ts

```diff
@@ -261,82 +267,152 @@ export class CddaData {
+
+    obj.__self = obj;
+    obj.__prevSelf = null;
+    obj.__mod = translate(obj.__mod, false, 1);
+
+    if (obj["copy-from"] && obj["copy-from"] === obj.id) {
+      const oldIndex = this._byType
+        .get(mappedType)!
+        .findIndex((x) => x.id === obj.id);
+      if (oldIndex !== -1) {
+        const oldObj = this._byType.get(mappedType)![oldIndex];
+        obj.__prevSelf = oldObj;
+      }
+    } else {
+      this._byType.get(mappedType)!.push(obj);
+    }
+
+    if (Object.hasOwnProperty.call(obj, "id")) {
+      if (!this._byTypeById.has(mappedType))
+        this._byTypeById.set(mappedType, new Map());
+      if (typeof obj.id === "string")
+        this._byTypeById.get(mappedType)!.set(obj.id, obj);
+      else if (Array.isArray(obj.id))
+        for (const id of obj.id) this._byTypeById.get(mappedType)!.set(id, obj);
+
+      // TODO: proper alias handling. We want to e.g. be able to collapse them in loot tables.
+      if (Array.isArray(obj.alias))
+        for (const id of obj.alias)
+          this._byTypeById.get(mappedType)!.set(id, obj);
+      else if (typeof obj.alias === "string")
+        this._byTypeById.get(mappedType)!.set(obj.alias, obj);
+    }
+    // recipes are id'd by their result
+    if (
+      (mappedType === "recipe" || mappedType === "uncraft") &&
+      Object.hasOwnProperty.call(obj, "result")
+    ) {
+      if (!this._byTypeById.has(mappedType))
+        this._byTypeById.set(mappedType, new Map());
+      const id =
+        obj.result +
+        (obj.variant && !obj.abstract ? "_" + obj.variant : "") +
+        (obj.id_suffix ? "_" + obj.id_suffix : "");
+      this._byTypeById.get(mappedType)!.set(id, obj);
+    }
+    if (
+      mappedType === "monstergroup" &&
+      Object.hasOwnProperty.call(obj, "name")
+    ) {
+      if (!this._byTypeById.has(mappedType))
+        this._byTypeById.set(mappedType, new Map());
+      const id = obj.name;
+      this._byTypeById.get(mappedType)!.set(id, obj);
+    }
+    if (Object.hasOwnProperty.call(obj, "abstract")) {
+      if (!this._abstractsByType.has(mappedType))
+        this._abstractsByType.set(mappedType, new Map());
+      this._abstractsByType.get(mappedType)!.set(obj.abstract, obj);
+    }
+
+    if (Object.hasOwnProperty.call(obj, "crafting_pseudo_item")) {
+      this._craftingPseudoItems.set(obj.crafting_pseudo_item, obj.id);
+    }
+
+    if (Object.hasOwnProperty.call(obj, "nested_mapgen_id")) {
+      if (!this._nestedMapgensById.has(obj.nested_mapgen_id))
+        this._nestedMapgensById.set(obj.nested_mapgen_id, []);
+      this._nestedMapgensById.get(obj.nested_mapgen_id)!.push(obj);
+    }
+  }
+
+  availableMods(): string[] {
+    return this._availableMods;
+  }
+
+  getModInfo(modId: string): { info: any; data: any[] } | undefined {
+    return this._rawMods[modId];
+  }
+
+  getEnabledMods(): string[] {
+    return this._enabledMods;
+  }
+
+  modEnabled() {
+    return this._enabledMods.length > 0;
```

Comment:
i think this is fine to be inlined at the call sites; no need for a dedicated helper function.